torch/csrc/jit/codegen/cuda/graph_fuser.cpp

```diff
@@ -751,12 +751,22 @@ struct CudaGraphFuser {
           // we scan this consumer again to perform the fusion
           return std::make_pair(consumer->reverseIterator(), true);
         }
-        auto fusion_group = tryFuse(consumer, producer);
+        auto fusion_group = tryFuse(consumer, producer->node());
         if (fusion_group) {
           // after fusion, consumer moves into a FusionGroup, so inputs is no
           // longer valid so we rescan the new FusionGroup for more fusions...
           return std::make_pair(fusion_group.value()->reverseIterator(), true);
         }
+        // horizontal fusion?!?!
+        for (const auto& u : producer->uses()) {
+          // only merge nodes before consumer
+          if (u.user->isBefore(consumer)) {
```

Comment:
Why if the user is before consumer? Is this just to avoid checking the same options over and over again? I assume isBefore returns false if u.user == consumer?

Reply:
Generally speaking, we fuse backwards (only look at nodes prior to self). If we have another sibling that is after consumer, then likely they have already checked whether they could be fused with consumer.
Yes, this is just to avoid some redundant checks.
Yes if u.user == consumer it returns false.